extensions/librdkafka/KafkaConnection.h

```diff
@@ -0,0 +1,217 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef NIFI_MINIFI_CPP_KAFKACONNECTION_H
+#define NIFI_MINIFI_CPP_KAFKACONNECTION_H
+
+#include <atomic>
+#include <mutex>
+#include <string>
+#include "core/logging/LoggerConfiguration.h"
+#include "core/logging/Logger.h"
+#include "rdkafka.h"
+#include "KafkaTopic.h"
+#include "KafkaPool.h"
+
+class KafkaConnectionKey {
+public:
+    std::string client_id_;
+    std::string brokers_;
+};
+
+struct KafkaConnectionComparator {
```

Comment:
I know this has been in the original code, but do you see any reason having a Comparator functor instead of KafkaConnectionKey having an `operator<`?